src/azure-account.api.d.ts

```diff
@@ -40,7 +43,7 @@ export interface AzureSession {
 	/**
 	 * The credentials object for azure-sdk-for-js modules https://github.com/azure/azure-sdk-for-js
 	 */
-	readonly credentials2: TokenCredentialsBase;
+	readonly credentials2: TokenCredentialsBase | AzureIdentityCredentialAdapter;
```

Comment:
There are some ADAL specific properties on [`TokenCredentialsBase`](https://github.com/Azure/ms-rest-nodeauth/blob/684d9825afe25fc907e8d163d19c0996702d0a70/lib/credentials/tokenCredentialsBase.ts#L16) that are not present on [`AzureIdentityCredentialAdapter`](https://github.com/Azure/ms-rest-js/blob/f6f0d92d79a1dfa8d92ff0891b88bc6b7a349e69/lib/credentials/azureIdentityTokenCredentialAdapter.ts#L17) so it could definitely break people. What do you think about creating a `credentials3` that's just type `AzureIdentityCredentialAdapter`? I don't think there's a way to fully avoid making a breaking change

Reply:
If we have to do a breaking change regardless, I would love if we can switch over to a versioned api, like this:

Reply:
To be clear, though, I think switching to a versioned API is probably best done in it's own PR.

Reply:
Tracking this here: https://github.com/microsoft/vscode-azure-account/issues/304